flomio/passninja-guestlist-gas
# Sheet streamlining: Fixes #55

# Changes
- Default to Y in config sheet
- Remove excess cells in Scanners and Config sheets (10 default): added function onPostSheetCreate + shrinkSheetRows which is responsible for automatically resizing/streamlining sheets based on config variables.  Also used different method which deletes all unused columns at once rather than one at a time (faster)
- All API external call events should be recorded in the EVENTS sheet when a non 200 response is received
- Message for non 200 external responses should not be "did you set your pass ninja credentials" but "Error: Check Events Sheet for details"
- Remove unfinished Form customization options columns
- Added system for default sheet column sizes
- Moved all code into `src/` directory to have a cleaner repository.
- Streamlined inefficient code in buildContactsSheet
- Added config hash comparison back in so we do not rebuild a sheet if user has not changed config variables and fixed the if statement that was incorrectly checking if it had been updated.
- Removed PRs from github action as it only should update master on a successful merge.
- Config cells under pass type are darkened to reduce possible confusion

# Verification
1.  Pull the latest changes from this branch: All files will be prefixed with src/ in the GAS project and will cause an error when pulling to update an older project using the original filenames, but if you refresh and pull once or twice after receiving the error it should be successful at overwriting/removing the old files.
1. Create a new GAS sheet (if you don't want to permanently hijack your Script Project that you have where you want it, start a new Script Project + deployment)
1. Set up a pass type with the proper fields and then `PassNinja->Create/Update Sheets from Config`
1. Observe that the Scanners/Config sheets are both much shorter and config is much more obviously laid out.
1. Mess up on setting your PassNinja credentials purposefully by entering nonsense values. (`PassNinja->Add Credentials->Set PassNinja Credentials`)
1. Try to create a pass (either using the live form, or manually entering contact information and selecting `PassNinja->Selected Row->Create/Update Pass`
1. Observe the error pop up with the proper error visible in the Toast notification and then check the Events sheet to see the detailed report (and response JSON body in the `Events.eventData Column`.  The eventType will show PassNinjaAPI and the response value.
1. Columns should be set to default widths now so you should be able to read most contents.
1. All GAS event times should now be localized instead of server time (PN events will still be whatever timezone they are designed to be displayed as).  Think Scan events and Error events.
